backend/internal/api/routes_services.go

```diff
@@ -25,13 +26,13 @@ func SetupRoutesWithServices(router *gin.Engine, nodeService *service.NodeServic
 		// Graph endpoints
 		api.GET("/graph", h.getGraph)
 		api.GET("/graph/viewport", h.getViewportNodes)
-		
+
 		// Node endpoints
 		api.GET("/nodes/:id", h.getNode)
 		api.GET("/nodes/:id/content", h.getNodeContent)
 		api.PUT("/nodes/:id/position", h.updateNodePosition)
 		api.GET("/nodes/search", h.searchNodes)
-		
+
 		// Vault management endpoints (for future use)
 		api.POST("/vault/parse", h.parseVault)
 		api.GET("/vault/status", h.getParseStatus)
```

Comment:
TODO(CL): Add middleware for parse endpoints:
1. Authentication middleware (these are admin operations)
2. Request ID middleware for tracking
3. Timeout middleware (different from other endpoints)
4. Audit logging middleware

Example:
```go
vaultGroup := api.Group("/vault")
vaultGroup.Use(RequireAdmin(), RequestIDMiddleware(), AuditLog())
vaultGroup.POST("/parses", h.parseVault)
vaultGroup.GET("/parses/:id", h.getParseStatus)
```